### ISSUE TITLE
bug: CtElement.getRoleInParent() and CtElement.getValueByRole(CtRole) are not consistent for catch with multple types

### ISSUE THREAD

**Issue by nharrand** [INRIA/spoon#1872]
When parsing 
```Java
	try {
		Object val = m.get(o);
		if (val != null) {
			return val.getClass().getCanonicalName();
		}
	} catch (IllegalArgumentException | IllegalAccessException e) {
		Launcher.LOGGER.error(e.getMessage(), e);
	}
```
the node corresponding to the type of `java.lang.IllegalArgumentException` has a role `CtRole.TYPE` in its parent `CtCatchVariable`, but if `getValueByRole(CtRole.TYPE)` is called on this parent, it returns the type `java.lang.Exception`. (Wheras it should return, if I understand correctly, a List containing both types `java.lang.IllegalArgumentException` and `java.lang.IllegalAccessException`)

**Comment by pvojtechovsky** [comment]
Hi Nicolas! Thanks for reporting that problem and defining test case. 

@monperrus, @surli it is a tricky case ... We probably have to add new CtRole#MULTI_TYPE

But note that there is a relation between new CtRole#MULTI_TYPE and existing CtRole#TYPE. The `type` is derived from `multiType` in this case... and because we do not provide access to derived properties using CtRole yet, we should - may be - remove CtRole#TYPE out of there??

WDYT?

**Comment by monperrus** [comment]
> we should - may be - remove CtRole#TYPE out of there

OK for me.

**Comment by monperrus** [comment]
closed per #1877
